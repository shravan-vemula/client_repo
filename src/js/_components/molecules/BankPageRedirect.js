import React from 'react'
import { Button } from '@material-ui/core';
import {useLocation} from 'react-router-dom';
import AnimationModalSuccess from 'js/_components/atoms/AnimationModalSuccess'



function BankPageRedirect() {
    const location = useLocation();
    const {url} = location.state;
    const [onUnloadWindow,setOnUnloadWindow] = React.useState(false);

    const handleClick = () =>{
       const myWindow = window.open(url, '_blank', 'location=yes,height=570,width=520,scrollbars=yes,status=yes',false)
  
       myWindow.addEventListener('unload',function(e){
           e.preventDefault();
            console.log(e)
            console.log("unloaded")
            setOnUnloadWindow(true);
              
        })

    }

    if(onUnloadWindow){
        return(
            <>
            <BankPageRedirect />
            <AnimationModalSuccess placeHolder={"Linking Bank"}/>    
            </>
        );
    }

    return (
        <>
    <div id="bank-selection-layout-1" className="bankSelectionLayout1">
        <div id="bank-selection-header" className="bankSelectionHeader">
            <div id="bank-selection-header-text-1"className="bankSelectionHeader1" >
                <p>Connect your account</p>
                </div>
            <div id="bank-selection-header-text-1"className="bankSelectionHeader2" >
                <p><b>Start by connecting your Chase Bank accounts with Intuit, the makers of Mint.</b><br /> <br />
                Go to the bank site to sign in and connect your accounts. Then weâ€™ll return you here and bring<br /> 
                in your transactions.</p>
            </div>  
            <div id="optional-selection" className="optionalSelection">
            <Button data-testId="continueButton" variant="outlined" color="primary" 
            onClick={handleClick}>
                Continue
            </Button>
            </div>  
        </div>
    </div>
    </>
    )
}

export default BankPageRedirect
