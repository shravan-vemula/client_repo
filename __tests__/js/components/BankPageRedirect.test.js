import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom'
import BankPageRedirect from 'js/_components/molecules/BankPageRedirect';
import SbiLogo from 'images/sbi.jpg';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
      pathname: "localhost:3000/example/path",
      state:{
        logo: '' ,
        url : 'https://retail.onlinesbi.com/retail/login.htm'
      }
    })
  }));

describe('BankPageRedirect renders properly', () => {
    const {getByTestId} = render(<Router><BankPageRedirect /></Router>);
    expect(getByTestId("continueButton")).toBe;
    fireEvent.click(getByTestId('continueButton'))
})


it("renders BankSelection correctly", () => {
    const wrapper = shallow(
        <BankPageRedirect /> 
    );
    expect(wrapper).toMatchSnapshot();
  });
