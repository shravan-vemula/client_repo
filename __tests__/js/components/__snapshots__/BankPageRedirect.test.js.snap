// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BankSelection correctly 1`] = `ShallowWrapper {}`;
